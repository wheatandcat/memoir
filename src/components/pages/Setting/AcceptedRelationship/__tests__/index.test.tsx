import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as Recoil from 'recoil';
import IndexPage, { Props } from '../';

const propsData = (): Props =>
  ({
    navigation: {
      setParams: jest.fn(),
      navigate: jest.fn(),
    },
    route: {
      params: {},
    },
  } as any);

describe('components/pages/Setting/AcceptedRelationship/index.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    jest.spyOn(Recoil, 'useRecoilValue').mockImplementation((): any => ({
      id: 'test-id',
      displayName: 'test-name',
    }));
    wrapper = shallow(<IndexPage {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
