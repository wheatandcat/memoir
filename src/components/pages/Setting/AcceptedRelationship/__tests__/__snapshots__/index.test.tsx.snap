// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Setting/AcceptedRelationship/index.tsx 正常にrenderすること 1`] = `
<Memo(Page)
  user={
    Object {
      "displayName": "test-name",
      "id": "test-id",
    }
  }
/>
`;
